aggregator/src/datastore.rs

```diff
@@ -3265,25 +3223,25 @@ ORDER BY id DESC
         let stmt = self
             .tx
             .prepare_cached(
-                r#"WITH aggregation_jobs_to_delete AS (
+                "WITH aggregation_jobs_to_delete AS (
                     SELECT id FROM aggregation_jobs
                     WHERE task_id = (SELECT id FROM tasks WHERE task_id = $1)
                       AND UPPER(client_timestamp_interval) <= $2
                       AND NOT EXISTS (
                           SELECT id FROM collect_jobs
                           WHERE aggregation_jobs.task_id = collect_jobs.task_id
-                            AND (aggregation_jobs.batch_identifier = collect_jobs.batch_identifier
-                              OR aggregation_jobs.batch_interval <@ collect_jobs.batch_interval))
+                            AND (aggregation_jobs.batch_id = collect_jobs.batch_identifier
+                             OR aggregation_jobs.client_timestamp_interval <@ collect_jobs.batch_interval))
```

Comment:
Should this switch to && as well?

Reply:
Good catch. To test this, I updated the unit tests to artificially extend the `client_timestamp_interval` by one second in each direction. I also spaced out the client timestamps to make sure there were no "false overlaps".